homeassistant/components/shopping_list/__init__.py

```diff
@@ -180,9 +183,22 @@ async def sort_list_service(call: ServiceCall) -> None:
     websocket_api.async_register_command(hass, websocket_handle_clear)
     websocket_api.async_register_command(hass, websocket_handle_reorder)
 
+    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)
+
     return True
 
 
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+
+    if unload_ok := await hass.config_entries.async_unload_platforms(
+        config_entry, PLATFORMS
+    ):
+        hass.data.pop(DOMAIN)
```

Comment:
The services and websocket command will fail if we remove the shopping list data.